src/crystal/system/unix/file.cr

```diff
@@ -57,6 +46,38 @@ module Crystal::System::File
     end
   end
 
+  # On some systems, the symbols `stat`, `fstat` and `lstat` are not part of the GNU
+  # shared library `libc.so` and instead provided by `libc_noshared.a`.
+  # That makes them unavailable for dynamic runtime symbol lookup via `dlsym`
+  # which we use for interpreted mode.
+  # See https://github.com/crystal-lang/crystal/issues/11157#issuecomment-949640034 for details.
+  # Linking against the internal counterparts `__xstat`, `__fxstat` and `__lxstat` directly
+  # should work in both interpreted and compiled mode.
+
+  def self.stat(path, stat)
```

Comment:
I'm somewhat confused that methods are being added to _crystal/system/unix/file.cr_ but not to _crystal/system/win32/file.cr_. It's like an interface and normally they mirror the same set of methods. So it would probably make a lot of sense to at least make these `private`. If you can actually do that with class methods; I don't remember.

Reply:
Yes, class methods can be private. But then you couldn't call  `fstat` from `FileDescriptor`.

System implementations are not limited to the set of platform-independent methods. They define whatever methods are useful for the specific implementation. In this case, it makes sense to abstract the different `*stat` function calls in methods. And I don't see a better place for put them then `Crystal::System::File`.